### PR TITLE
Common-token-manager module. Spring config. Tests.

### PR DESCRIPTION
**Pull request by aptasinskij**

### CODE REVIEW

File: basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Date;
+
+@Component
+public class JwtTokenManager implements TokenManager {
+
+    private String UNLIMITED = "AUdl2nrHDBqm7vzcwu72QjZUsBsFNQ";
+    private String SHORT = "cIrclShaikXm5MSDk5EvlKnzXwbmsS";
+
+    public String getUnlimitedToken(String userId, String deviceId) {
+        if (deviceId == null || deviceId.isEmpty()) return null;
```
Comment (Dominys) [inline basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java]: @aptasinskij Use apache StringUtils for such checks. Next - if we can not create token - than we should throw exception

File: basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                .compact();
+    }
+
+    public String getShortToken(String unlimitedToken) {
+        if (unlimitedToken == null || unlimitedToken.isEmpty()) return null;
+        Claims claims;
+        try {
+            claims = Jwts.parser().setSigningKey(UNLIMITED).parseClaimsJws(unlimitedToken).getBody();
+        } catch (Exception ex){
+            return null;
```
Comment (Dominys) [inline basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java]: @aptasinskij null is not a valid response if we have invalid token.

File: basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.impl.DefaultClock;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+public class JwtTokenManager implements TokenManager {
```
Comment (Dominys) [inline basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/JwtTokenManager.java]: @aptasinskij Your code is hard readable. Make helper methods and make code readable.

File: basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/keyproducer/JwtKeyProducer.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.softgroup.common.tokenmanager.keyproducer;
+
+import com.softgroup.common.tokenmanager.TokenType;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class JwtKeyProducer implements KeyProducer {
```
Comment (Dominys) [inline basis/common-token-manager/src/main/java/com/softgroup/common/tokenmanager/keyproducer/JwtKeyProducer.java]: @aptasinskij Good idea with key producer. But I will tell you how to make even better. Make additional module - PropertiesProvider. That module will provide settings to other modules.

**Comment by aptasinskij** [comment]
@Dominys, please, review the code)

File: basis/common-properties-provider/src/main/java/com/softgroup/common/propertiesprovider/PropertiesProvider.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.softgroup.common.propertiesprovider;
+
+public interface PropertiesProvider {
+    <T> T getProperty(Class<T> tClass, String key);
```
Comment (Dominys) [inline basis/common-properties-provider/src/main/java/com/softgroup/common/propertiesprovider/PropertiesProvider.java]: @aptasinskij Properties are usually simple types like string, integer etc. No need to make generic methods for that. 
Usually properties holder has methods like "getStringValue", "getIntValue" and so on

Comment (aptasinskij) [inline basis/common-properties-provider/src/main/java/com/softgroup/common/propertiesprovider/PropertiesProvider.java]: @Dominys, even org.apache.commons.configuration2.Configuration class (that I wrapped in) has list of methods like `getArray()`, `getBigDecimal`, `getFloat()`, `getString()` and so on (see [JavaDoc](https://commons.apache.org/proper/commons-configuration/apidocs/index.html)), so I decide that in some cases it my be usefull to have a possibility to provide different configuration values, including float number etc. Base on that i decide not to duplicate all methods from wraped class but provide universal methods for getting any property value.

**Pull request merged**